tests/backwards_compatibility_tests/scripts/determine_to_version.py

```diff
@@ -0,0 +1,45 @@
+import os
+import subprocess
+import sys
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../../../src'))
+from marqo.version import __version__
+import semver
+
+def determine_to_version(run_commit_hash: str, current_marqo_version: str):
+    """
+    This function determines the to_version.
+    It does so by looking at version.py file. However there can be times where active development is going on and
+    Marqo developers have not yet updated the version.py file with the next (i.e To be released) version. In such cases, we need to
+    determine the to version by looking at the latest tag in the git repository.
+
+    If for version v, tag t exists, then it could mean that version v is already released and the developers are working on version v+1.
+    We determine if this is the case by comparing commit hash of tag t and commit hash of the github workflow run. If they're different, we can
+    conclude that version v is already released, and to test backwards compatibility we need to test against version v as well. Thus we set to_version = v+1.
+
+    If the commit hash of tag t and commit hash of the github workflow run are the same, then we can conclude that this may be a re-run. Similar to this case,
+    if the tag t for version v doesn't exist yet, we can determine that version v is the upcoming To be released version. In this case we set to_version = v.
+    """
+    tag = subprocess.check_output(["git", "tag", "--list", f"{current_marqo_version}"],
+                                   text=True).splitlines() #Determine if tags exist for current_marqo_version picked from version.py file
+    if tag: #If tag already exists for the current_marqo_version, it means that this version is already released and we possibly have not bumped up the version yet, thus we need to treat this commit as commit of the next version.
```

Comment:
If I understand correctly, that means we are working towards a release right, which is the most common case that we merge a PR?

Reply:
Yes